pages/fringe-events.jsx

```diff
@@ -1,21 +1,247 @@
+    var field = fieldType !== "textarea" ? <input type={this.props.fieldType || "text"}
+                                                  name={this.props.name}
+                                                  id={fieldId}
+                                                  placeholder={this.props.placeholder}
+                                                  valueLink={this.linkField(this.props.name)} />
+                                         : <textarea type={this.props.fieldType || "text"}
+                                                     name={this.props.name}
+                                                     id={fieldId}
+                                                     placeholder={this.props.placeholder}
+                                                     valueLink={this.linkField(this.props.name)}
+                                                     rows="3" />;
+    return (
+      <fieldset>
+        <label htmlFor={fieldId}>
+          {this.props.label}
+          {this.props.required ? <span>*</span> : null}
+        </label>
+        {this.props.additionalText ? <p className="additional-text">{this.props.additionalText}</p> : null }
+        {field}
+        {this.props.exampleValue ? <p className="example-value">{this.props.exampleValue}</p> : null }
+        {this.props.label === "Privacy Policy" ? <span>I’m okay with Mozilla handling this info as you explain in your <a href="https://www.mozilla.org/privacy/">privacy policy</a>. Please note: the information in this form will be submitted via a private Google form.</span>
+                                               : null
+        }
+        <ErrorMessage field={this.props.name} />
+      </fieldset>
+    );
+  }
+});
+
+var FringeEventForm = Formation.CreateForm({
+  getSchema: function () {
+    return {
+      "name": {
+        required: true,
+        label: 'Name of Fringe Event'
+      },
+      "time": {
+        required: true,
+        label: 'Date and Time',
+        validations: this.dateTimeValidator,
+        placeholder: "dd/mm/yyyy --:-- --",
+        exampleValue: "Example: 31/10/2015 03:30 PM"
+      },
+      "location": {
+        required: true,
+        label: 'Event Location',
+        additionalText: "Please add full address here, including post code"
+      },
+      "description": {
+        required: true,
+        label: 'Description of your Event',
+        validations: this.wordCountValidator,
+        additionalText: "Maximum 100 words",
+        fieldType: "textarea"
+      },
+      "link": {
+        required: false,
+        label: 'Link to register for the event if available',
+        validations: Validator.url()
+      },
+      "privacy": {
+        required: true,
+        label: 'Privacy Policy',
+        fieldType: "checkbox"
+      }
+    };
+  },
+  wordCountValidator: function(value) {
+    return value.split(" ").length > 100 ? "Maximum length for description is 100 words." : false;
```

Comment:
given that we tested it with a 10000 letter string, it's a good idea if we also cut this off at "max 100 words / 1000 characters"